src/dune_rules/simple_rules.ml

```diff
@@ -117,8 +118,14 @@ let user_rule sctx ?extra_bindings ~dir ~expander (rule : Rule.t) =
     | Alias_only name ->
       let alias = Alias.make ~dir name in
       let* locks = interpret_locks ~expander rule.locks in
+      let patch_back_source_tree =
+        match rule.mode with
+        | Patch_back_source_tree -> true
```

Comment:
Maybe worth creating a helper function for this, as the same pattern match appears one more time above.

Reply:
Where else does it appear?

Reply:
In the call to `Sandbox.create`.

Reply:
That was in a previous version of the PR. `Sandbox.create` now takes a boolean directly

Reply:
I meant that call site of `Sandbox.create` does the same pattern match as we do here:

```ocaml
match rule_mode with
    | Patch_back_source_tree -> true
    | _ -> false
```
So, this could be a function in `Rule.Mode` but it's not super important of course.